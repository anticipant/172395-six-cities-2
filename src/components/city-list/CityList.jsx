import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

class CityList extends PureComponent {
  constructor(props) {
    super(props);
    this.cityClickHandler = this.cityClickHandler.bind(this);
  }

  cityClickHandler(evt) {
    const cityName = evt.currentTarget.querySelector(`span`).innerHTML;
    this.props.selectCity(cityName);
  }

  render() {
    const {cities, name} = this.props;
    return cities.map((it) => {
      return (
        <li
          key={`${it}`}
          onClick={this.cityClickHandler}
          className="locations__item"
        >
          <a className={`locations__item-link tabs__item ${name === it ? `tabs__item--active` : ``}`} href="#">
            <span>{it}</span>
          </a>
        </li>
      );
    });
  }
}

CityList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  name: PropTypes.string.isRequired,
  selectCity: PropTypes.func.isRequired,
};

export default CityList;
