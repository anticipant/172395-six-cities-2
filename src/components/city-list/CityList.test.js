import React from 'react';
import renderer from 'react-test-renderer';

import CityList from "./CityList";

it(`CityList correctly renders after relaunch`, () => {
  const tree = renderer.create(<CityList
    cities={[`minsk`]}
    name={`minsk`}
    selectCity={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
