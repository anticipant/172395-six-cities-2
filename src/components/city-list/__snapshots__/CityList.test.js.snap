// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList correctly renders after relaunch 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
  >
    <span>
      minsk
    </span>
  </a>
</li>
`;
